### PR TITLE
Don't regenerate jwt secret

### PR DESCRIPTION
**Pull request by sanjid133**

### CODE REVIEW

**Review by zeripath — APPROVED** [review]

**Review by adelowo — APPROVED** [review]

File: modules/setting/setting.go [hunk 1]
```diff
@@ -728,10 +728,12 @@ func NewContext() {
 	}
 
 	if OAuth2.Enable {
+		OAuth2.JWTSecretBase64 = Cfg.Section("oauth2").Key("JWT_SECRET").String()
+
 		OAuth2.JWTSecretBytes = make([]byte, 32)
 		n, err := base64.RawURLEncoding.Decode(OAuth2.JWTSecretBytes, []byte(OAuth2.JWTSecretBase64))
 
-		if err != nil || n != 32 {
+		if len(OAuth2.JWTSecretBase64) == 0 && (err != nil || n != 32) {
```
Comment (zeripath) [inline modules/setting/setting.go:736]:
```suggestion
		if err != nil || n != 32 {
```

We still want this to be regenerated if JWTSecretBase64 is not valid

**Review by zeripath — CHANGES_REQUESTED** [review]
See above

**Comment by zeripath** [comment]
Hmm actually looking again is this actually even necessary at all?

It looks like this should work by default.

File: modules/setting/setting.go [hunk 2]
```diff
@@ -728,10 +728,12 @@ func NewContext() {
 	}
 
 	if OAuth2.Enable {
+		OAuth2.JWTSecretBase64 = Cfg.Section("oauth2").Key("JWT_SECRET").String()
```
Comment (lunny) [inline modules/setting/setting.go:731]: This has been read from line 725.

**Review by lunny — CHANGES_REQUESTED** [review]

**Comment by zeripath** [comment]
Yeah @lunny I agree, I approved too quickly. I think this pr might actually be unnecessary. The change to the check line is wrong and introduced a security bug. The reloading of the secret seems also unnecessary.

@sanjid133 what was the reason behind this pr?

**Comment by sanjid133** [comment]
@zeripath If I mount the app.ini file as a secret in only read mode regenerating this token is creating an error.

**Comment by lunny** [comment]
@sanjid133 that's another issue. I will close this PR since it will not resolve anything.

**Pull request closed without merging**